### DISCUSSION TITLE
zh/guide/install/source

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#102] [Announcements]
# zh/guide/install/source

此步骤仅适用于需要自行修改源代码的用户。 不明白的请跳过。 环境准备 首先，你需要一个有git，`nodejs，pnpm，golang>=1.18，gcc的环境 构建前端 使用 git clone --recursive https://github.com/alist-org/alist-web.git 克隆前端 ，执行 pnpm install &...

https://alist.nn.ci/zh/guide/install/source.html

**Comment by clockworkchen** [comment]
alist-web仓库只有英文语言包吗？是不是中文的没有提交呢。

**Reply by anwen-anyi** [reply]
简体中文语言包：https://crowdin.com/project/alist/zh-CN

**Comment by STEVENTAN100** [comment]
请问如果要修改可同时进行复制的文件次数为10，请问在源码的哪个位置修改呢？

**Comment by oOTnTOo** [comment]
编译倒是完成了，但是Web端是运行起来是无限转圈啊

**Reply by oOTnTOo** [reply]
好吧，知道了，简体中文还要再运行下 i18n.mjs 来生成TS文件才能用……，成功了，多谢

**Comment by jldhrj** [comment]
最后编译过程中报错提示：
$ /f/alisst/alist/run.sh
/f/alisst/alist/run.sh: line 3: go: command not found
/f/alisst/alist/run.sh: line 17: go: command not found

windows10下进行的操作，请大佬指点，一直说go没有此命令
appName="alist"
builtAt="$(date +'%F %T %z')"
goVersion=$(go version | sed 's/go version //')
gitAuthor=$(git show -s --format='format:%aN <%ae>' HEAD)
gitCommit=$(git log --pretty=format:"%h" -1)
version=$(git describe --long --tags --dirty --always)
webVersion=$(wget -qO- -t1 -T2 "https://api.github.com/repos/alist-org/alist-web/releases/latest" | grep "tag_name" | head -n 1 | awk -F ":" '{print $2}' | sed 's/\"//g;s/,//g;s/ //g')
ldflags="\
-w -s \
-X 'github.com/alist-org/alist/v3/internal/conf.BuiltAt=$builtAt' \
-X 'github.com/alist-org/alist/v3/internal/conf.GoVersion=$goVersion' \
-X 'github.com/alist-org/alist/v3/internal/conf.GitAuthor=$gitAuthor' \
-X 'github.com/alist-org/alist/v3/internal/conf.GitCommit=$gitCommit' \
-X 'github.com/alist-org/alist/v3/internal/conf.Version=$version' \
-X 'github.com/alist-org/alist/v3/internal/conf.WebVersion=$webVersion' \
"
go build -ldflags="$ldflags" .

------------------------------------------------------

go已经安装过了，环境边变量也是设置过
C:\Users\Administrator>go version
go version go1.21.0 windows/amd64
-------------------------------------------------------
gcc：
C:\Users\Administrator>gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=f:/mingw/bin/../libexec/gcc/mingw32/6.3.0/lto-wrapper.exe
Target: mingw32
Configured with: ../src/gcc-6.3.0/configure --build=x86_64-pc-linux-gnu --host=mingw32 --target=mingw32 --with-gmp=/mingw --with-mpfr --with-mpc=/mingw --with-isl=/mingw --prefix=/mingw --disable-win32-registry --with-arch=i586 --with-tune=generic --enable-languages=c,c++,objc,obj-c++,fortran,ada --with-pkgversion='MinGW.org GCC-6.3.0-1' --enable-static --enable-shared --enable-threads --with-dwarf2 --disable-sjlj-exceptions --enable-version-specific-runtime-libs --with-libiconv-prefix=/mingw --with-libintl-prefix=/mingw --enable-libstdcxx-debug --enable-libgomp --disable-libvtv --enable-nls
Thread model: win32
gcc version 6.3.0 (MinGW.org GCC-6.3.0-1)

**Comment by xiangdeyue** [comment]
编译ios版本的alist后，运行报错：
iPhone:~/alist root# ./alist start
SIGSEGV: segmentation violation
PC=0x181e82eb8 m=0 sigcode=2
signal arrived during cgo execution

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x105c7ed44, 0x13006adf8)
	/usr/local/go/src/runtime/cgocall.go:157 +0x44 fp=0x13006adc0 sp=0x13006ad80 pc=0x104a51e24
github.com/shoenig/go-m1cpu._Cfunc_initialize()
	_cgo_gotypes.go:143 +0x30 fp=0x13006adf0 sp=0x13006adc0 pc=0x10586b430
github.com/shoenig/go-m1cpu.init.0()
	/home/gopath/pkg/mod/github.com/shoenig/go-m1cpu@v0.1.6/cpu.go:148 +0x1c fp=0x13006ae00 sp=0x13006adf0 pc=0x10586b46c
runtime.doInit1(0x107897570)
	/usr/local/go/src/runtime/proc.go:6740 +0xd8 fp=0x13006af30 sp=0x13006ae00 pc=0x104a96738
runtime.doInit(...)
	/usr/local/go/src/runtime/proc.go:6707
runtime.main()
	/usr/local/go/src/runtime/proc.go:249 +0x374 fp=0x13006afd0 sp=0x13006af30 pc=0x104a88824
runtime.goexit()
	/usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x13006afd0 sp=0x13006afd0 pc=0x104abde84

goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/usr/local/go/src/runtime/proc.go:398 +0xc8 fp=0x13006bf90 sp=0x13006bf70 pc=0x104a88bc8
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:404
runtime.forcegchelper()
	/usr/local/go/src/runtime/proc.go:322 +0xb8 fp=0x13006bfd0 sp=0x13006bf90 pc=0x104a88a28
runtime.goexit()
	/usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x13006bfd0 sp=0x13006bfd0 pc=0x104abde84
created by runtime.init.6 in goroutine 1
	/usr/local/go/src/runtime/proc.go:310 +0x24

goroutine 3 [GC sweep wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/usr/local/go/src/runtime/proc.go:398 +0xc8 fp=0x13006cf60 sp=0x13006cf40 pc=0x104a88bc8
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:404
runtime.bgsweep(0x0?)
	/usr/local/go/src/runtime/mgcsweep.go:280 +0xa0 fp=0x13006cfb0 sp=0x13006cf60 pc=0x104a73360
runtime.gcenable.func1()
	/usr/local/go/src/runtime/mgc.go:200 +0x28 fp=0x13006cfd0 sp=0x13006cfb0 pc=0x104a67c98
runtime.goexit()
	/usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x13006cfd0 sp=0x13006cfd0 pc=0x104abde84
created by runtime.gcenable in goroutine 1
	/usr/local/go/src/runtime/mgc.go:200 +0x6c

goroutine 4 [GC scavenge wait]:
runtime.gopark(0x13007a000?, 0x106794de0?, 0x1?, 0x0?, 0x1300031e0?)
	/usr/local/go/src/runtime/proc.go:398 +0xc8 fp=0x13006df50 sp=0x13006df30 pc=0x104a88bc8
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:404
runtime.(*scavengerState).park(0x1079c4c40)
	/usr/local/go/src/runtime/mgcscavenge.go:425 +0x5c fp=0x13006df80 sp=0x13006df50 pc=0x104a70bcc
runtime.bgscavenge(0x0?)
	/usr/local/go/src/runtime/mgcscavenge.go:653 +0x44 fp=0x13006dfb0 sp=0x13006df80 pc=0x104a71124
runtime.gcenable.func2()
	/usr/local/go/src/runtime/mgc.go:201 +0x28 fp=0x13006dfd0 sp=0x13006dfb0 pc=0x104a67c38
runtime.goexit()
	/usr/local/go/src/runtime/asm_arm64.s:1197 +0x4 fp=0x13006dfd0 sp=0x13006dfd0 pc=0x104abde84
created by runtime.gcenable in goroutine 1
	/usr/local/go/src/runtime/mgc.go:201 +0xac

r0      0x0
r1      0x0
r2      0x60
r3      0x48
r4      0x507
r5      0x0
r6      0x0
r7      0x16b3ba244
r8      0x4d8381aa5d270085
r9      0x4d8381aa5d270085
r10     0x15
r11     0x7f
r12     0x15
r13     0x59
r14     0x455
r15     0xb2
r16     0x181e82ea8
r17     0xfbf80
r18     0x0
r19     0x0
r20     0x0
r21     0x0
r22     0x16b3bacb8
r23     0x0
r24     0x0
r25     0x0
r26     0x107897578
r27     0x820
r28     0x1079c5660
r29     0x16b3baa90
lr      0x105c7e71c
sp      0x16b3baa70
pc      0x181e82eb8
fault   0x0

**Comment by huolanyun** [comment]
linux编译  https://www.yuque.com/anwenya/alist/glqlhu#dV53r
附解决方法：
执行pnpm install && pnpm run build

出现 带这个的错误  error (CERT_HAS_EXPIRED)

可以执行下面命令取消SSL验证后再去执行（pnpm install && pnpm run build）
npm config set strict-ssl false
